Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlFulltextDataParameterDefinitionDecoratorTest.cs

```diff
@@ -0,0 +1,172 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Parameters;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.Parameters;
+
+[TestFixture]
+public class SqlFulltextDataParameterDefinitionDecoratorTest
+{
+  [Test]
+  [TestCaseSource(nameof(GetDummyValues))]
+  public void MaxSize_AnsiString_SizeSetTo8000 (object value)
+  {
+    var dbCommandStub = new Mock<IDbCommand>();
+
+    var dataParameterStub = new Mock<IDbDataParameter>();
+    dataParameterStub.Setup(_ => _.Value).Returns(value);
+    dataParameterStub.Setup(_ => _.DbType).Returns(DbType.AnsiString);
+    dataParameterStub.SetupProperty(_ => _.Size);
+    dataParameterStub.Object.Size = -1;
+
+    var parameterDefinitionMock = new Mock<IDataParameterDefinition>(MockBehavior.Strict);
+    parameterDefinitionMock.Setup(_ => _.CreateDataParameter(dbCommandStub.Object, value)).Returns(dataParameterStub.Object);
+
+    var decorator = new SqlFulltextDataParameterDefinitionDecorator(parameterDefinitionMock.Object);
+
+    var result = decorator.CreateDataParameter(dbCommandStub.Object, value);
+
+    Assert.That(result, Is.SameAs(dataParameterStub.Object));
+    Assert.That(result.Value, Is.EqualTo(value));
+    Assert.That(result.DbType, Is.EqualTo(DbType.AnsiString));
+    Assert.That(result.Size, Is.EqualTo(8000));
+  }
+
+  private static IEnumerable<object> GetDummyValues ()
+  {
+    yield return "Dummy";
+    yield return "Dummy".ToCharArray();
+  }
+
+  private static IEnumerable<object> GetValuesLength8001 ()
+  {
+    yield return new string('x', 8001);
+    yield return new string('x', 8001).ToCharArray();
+  }
+
+  private static IEnumerable<object> GetValuesLength4001 ()
+  {
+    yield return new string('x', 4001);
+    yield return new string('x', 4001).ToCharArray();
+  }
```

Comment:
please move to top or bottom (or leave right in front of the individual tests. right now it's randomly in the middle.